File: tests/factories/FabricaPaginas.ts
import { Page } from '@playwright/test';

import { PagInicio } from '@pom/PagInicio';
import { PagClientes } from '@pom/PagClientes';
import { PagCertificaciones } from '@pom/PagCertificaciones';
import { PagCursosEtiquetas } from '@pom/PagCursosEtiquetas';
import { PagCursosGrafico } from '@pom/PagCursosGrafico';
import { PagExperiencia } from '@pom/PagExperiencia';
import { PagJardin } from '@pom/PagJardin';
import { PagMetodo } from '@pom/PagMetodo';

export class FabricaPaginas {
  constructor(private readonly page: Page) {}

  crearInicio(): PagInicio {
    return new PagInicio(this.page);
  }

  crearClientes(): PagClientes {
    return new PagClientes(this.page);
  }

  crearCertificaciones(): PagCertificaciones {
    return new PagCertificaciones(this.page);
  }

  crearCursosEtiquetas(): PagCursosEtiquetas {
    return new PagCursosEtiquetas(this.page);
  }

  crearCursosGrafico(): PagCursosGrafico {
    return new PagCursosGrafico(this.page);
  }

  crearExperiencia(): PagExperiencia {
    return new PagExperiencia(this.page);
  }

  crearJardin(): PagJardin {
    return new PagJardin(this.page);
  }

  crearMetodo(): PagMetodo {
    return new PagMetodo(this.page);
  }
}

File: playwright-dotnet/PlaywrightDotnet.Tests/Factories/FabricaPaginas.cs
using Microsoft.Playwright;
using PlaywrightDotnet.Tests.Pom;

namespace PlaywrightDotnet.Tests.Factories
{
    public class FabricaPaginas
    {
        private readonly IPage page;

        public FabricaPaginas(IPage page)
        {
            this.page = page;
        }

        public PagInicio CrearInicio() => new PagInicio(page);
        public PagClientes CrearClientes() => new PagClientes(page);
        public PagCertificaciones CrearCertificaciones() => new PagCertificaciones(page);
        public PagCursosEtiquetas CrearCursosEtiquetas() => new PagCursosEtiquetas(page);
        public PagCursosGrafico CrearCursosGrafico() => new PagCursosGrafico(page);
        public PagExperiencia CrearExperiencia() => new PagExperiencia(page);
        public PagJardin CrearJardin() => new PagJardin(page);
        public PagMetodo CrearMetodo() => new PagMetodo(page);
    }
}
